src/components/appBar/AppBar.tsx

```diff
@@ -0,0 +1,55 @@
+import React from 'react';
+import { StatusBar, StyleSheet, View, ViewStyle } from 'react-native';
+import { Appbar as BaseAppBar } from 'react-native-paper';
+import { useTheme } from '../../styles/themes';
+import { Menu, MenuProps } from '../menu/Menu';
+import { Headline } from '../typography/Headline';
+
+interface Props extends MenuProps {
```

Comment:
Instead of exposing the Menu props interface through the AppBar props, would it be possible to have a more declarative approach like:

```html
<!-- Default -->

<AppBar title={...} iconName={...} onPress={...}/>

<!-- Or with a menu -->

<AppBar title={...} iconName={...} onPress={...}>
    <AppBar.Menu options={...} ..otherMenuProps>
<AppBar/>

```

Reply:
To clarify:

In the design system, the "Menu" in [the AppBar is optional](https://www.figma.com/file/MD0xdHA7E93GKAwkSRpduK/SW-mobile-app---Design-Kit?node-id=21%3A140&t=QhycZDswtVjJVajX-4)

Currently in the implementation, we toggle the menu display with [the props "menuVisible"](https://github.com/ZeroGachis/smartway-react-native-ui/pull/28/commits/638acce52ddc85386ca3a7da90e8b83d99fb7126#diff-c6c41a63edc421c771a2876c2106fe2fae5c4b89ed27c9a6468c3eb9c38f7933R21) and you expose all menu props "menuOptions", "menuContainerStyle", "onMenuPress", etc...

I think we could have a more clean API by allowing the AppBar component to accept an optional Menu component as its children.

This would allow us no define the "default" variant of the AppBar like this :

```html
<AppBar title={...} iconName={...} onPress={...}/>
```

And if we want to display a Menu in a given page, we could just do:

```html
<AppBar title={...} iconName={...} onPress={...}> 
   <Menu options={...} ..otherMenuProps>
<AppBar/>
```

The same way react-native-paper allows you to define specific children to its AppBar (AppBar.Content, AppBar.Action etc...)
Cf: https://callstack.github.io/react-native-paper/docs/components/Appbar/